spark/src/main/scala/ai/chronon/spark/Analyzer.scala

```diff
@@ -302,6 +302,35 @@ class Analyzer(tableUtils: TableUtils,
       (analysis, leftDf)
     }
 
+    val timestampChecks = runTimestampChecks(leftDf)
+    if (!timestampChecks.contains("noTsColumn")) {
+      // do timestamp checks
+      assert(
+        timestampChecks("notNullCount") != "0",
+        s"""[ERROR]: Join validation failed.
+           | Please check that left source has non-null timestamps.
+           | check notNullCount: ${timestampChecks("notNullCount")}
+           | """.stripMargin
+      )
+      assert(
+        timestampChecks("badRangeCount") == "0",
+        s"""[ERROR]: Join validation failed.
+           | Please check that left source has valid epoch millisecond timestamps.
+           | badRangeCount: ${timestampChecks("badRangeCount")}
+           | """.stripMargin
+      )
+
+      logger.info(s"""ANALYSIS TIMESTAMP completed for join/${name}.
+                     |check notNullCount: ${timestampChecks("notNullCount")}
+                     |check badRangeCount: ${timestampChecks("badRangeCount")}
+                     |""".stripMargin)
+
+    } else {
+      logger.info(s"""ANALYSIS TIMESTAMP completed for join/${name}.
+                     |check TsColumn: ${timestampChecks("noTsColumn")}
+                     |""".stripMargin)
+    }
```

Comment:
nit: I think this validation part can either be integrated into `runTimestampChecks()` or be refactored as an separate method, say `validateTimestamChecks(timestampChecks, "join")`

Reply:
done.